hoot-services/src/main/java/hoot/services/controllers/grail/GrailResource.java

```diff
@@ -186,6 +190,12 @@ public Response createDifferential(@Context HttpServletRequest request,
             GrailParams reqParams,
             @QueryParam("DEBUG_LEVEL") @DefaultValue("info") String debugLevel) {
 
+        Users user = Users.fromRequest(request);
+        HashMap privileges = ((HashMap) user.getPrivileges());
+        if(privileges == null || privileges.get("advanced") != "true") {
+            return Response.status(Response.Status.FORBIDDEN).type(MediaType.TEXT_PLAIN).entity("You do not have access to this operation.").build();
```

Comment:
I know we were going back and forth about returning a Response and throwing a WebAppException.
Would it be better to put this advanced user check logic in a method that throws said exception and that way make the code DRY?

Reply:
I thought about throws, but then in the locations that call the code I would have to wrap it in a try catch statement or else the throws will fill the tomcat log with an ugly stack trace and i thought we'd want to avoid that one?